llvm/lib/Support/CrashRecoveryContext.cpp

```diff
@@ -13,8 +13,17 @@
 #include "llvm/Support/Signals.h"
 #include "llvm/Support/thread.h"
 #include <cassert>
+#if !defined(__wasi__)
+#include <csignal>
+#endif
+#if LLVM_ENABLE_THREADS
```

Comment:
This is unfortunate; there's no reason (in theory) that mutex shouldn't work fine in single-threaded mode.

Reply:
I think it actually does work, since I ended up building LLVM always with the `wasm32-wasip1-threads` target, and this is a remnant of an earlier patch.